### PR TITLE
Improve Star Wars Example 

### PR DESCRIPTION
**Pull request by ThisaruGuruge**
## Purpose

Fixes: [#4205](https://github.com/ballerina-platform/ballerina-standard-library/issues/4205)

## Checklist
- [x] Linked to an issue
- [ ] ~Updated the changelog~
- [ ] ~Added tests~
- [ ] ~Updated the spec~
- [ ] ~Checked native-image compatibility~

### CODE REVIEW

File: examples/starwars/tests/websocket_utils.bal [hunk 1]
```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2022, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
```
Comment (DimuthuMadushan) [inline examples/starwars/tests/websocket_utils.bal:1]:
```suggestion
// Copyright (c) 2023, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
```

File: examples/starwars/tests/websocket_utils.bal [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    check wsClient->writeMessage({'type: WS_INIT});
+}
+
+isolated function initiateGraphqlWsConnection(websocket:Client wsClient) returns websocket:Error? {
+    check sendConnectionInitMessage(wsClient);
+    check validateConnectionAckMessage(wsClient);
+}
+
+isolated function sendSubscriptionMessage(websocket:Client wsClient, string document, string id = "1",
+        json? variables = {}, string? operationName = ()) returns websocket:Error? {
```
Comment (DimuthuMadushan) [inline examples/starwars/tests/websocket_utils.bal:43]: it seems the formatting is incorrect here.

Comment (ThisaruGuruge) [reply examples/starwars/tests/websocket_utils.bal:43]: This is how the Ballerina formatter formats the code.

File: examples/starwars/service.bal [hunk 3]
```diff
@@ -58,10 +70,12 @@ service /graphql on new graphql:Listener(9000) {
     resource function get characters(string[] idList) returns Character?[] {
         Character[] characters = [];
```
Comment (MohamedSabthar) [inline examples/starwars/service.bal:71]: Shouldn't this be `Character?[]`

File: examples/starwars/service.bal [hunk 4]
```diff
@@ -58,10 +70,12 @@ service /graphql on new graphql:Listener(9000) {
...
-            if ds:humanTable[id] is ds:HumanRecord {
-                characters.push(new Human(<ds:HumanRecord> ds:humanTable[id]));
-            } else if ds:droidTable[id] is ds:DroidRecord {
-                characters.push(new Droid(<ds:DroidRecord> ds:droidTable[id]));
+            if ds:humanTable.hasKey(id) {
+                characters.push(new Human(ds:humanTable.get(id)));
+            } else if ds:droidTable.hasKey(id) {
+                characters.push(new Droid(ds:droidTable.get(id)));
+            } else {
+                characters.push();
```
Comment (MohamedSabthar) [inline examples/starwars/service.bal:78]: don't we have to push nil here ?

Comment (ThisaruGuruge) [reply examples/starwars/service.bal]: Fixed with 1d4f453047b0cfb1db1e0570a17cf4617994110f

**Review by MohamedSabthar — APPROVED** [review]

**Review by DimuthuMadushan — APPROVED** [review]

**Pull request merged**